extensions/azurecore/src/azurecore.d.ts

```diff
@@ -355,7 +355,22 @@ declare module 'azurecore' {
 			getTreeDataProvider(): IAzureResourceTreeDataProvider;
 		}
 
-		export interface IAzureResourceTreeDataProvider extends TreeDataProvider<IAzureResourceNode> {
+		export interface IAzureResourceTreeDataProvider {
+			 /**
+			  * Gets the root tree item nodes for this provider - these will be used as
+			  * direct children of the Account node in the Azure tree view.
+			  */
+			getRootChildren(): Promise<azdata.TreeItem[]>;
```

Comment:
Splitting these up wasn't my favorite thing - keeping the behavior of getChildren with undefined being the root nodes does make sense. But it got too complicated with having two different types that could be returned and so I'm splitting them up for now. I want to revisit potentially changing this again though once I've had a chance to look at it more to see if we can come up with a better pattern (like potentially having the sub values be optional and filling in from root node if they aren't provided).